src/test/java/seedu/address/logic/parser/CheckinCommandParserTest.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.commands.CommandTestUtil.DATE_END_DESC_AMY;
+import static seedu.address.logic.commands.CommandTestUtil.DATE_START_DESC_AMY;
+import static seedu.address.logic.commands.CommandTestUtil.ROOM_DESC_001;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_ROOM_NUMBER_001;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import org.junit.Test;
+
+import seedu.address.logic.commands.CheckInCommand;
+import seedu.address.model.room.RoomNumber;
+
+/**
+ * As we are only doing white-box testing, our test cases do not cover path variations
+ * outside of the CheckInCommand code. For example, inputs "1", "01" , and "0001" take the
+ * same path through the CheckInCommand, and therefore we test only one of
+ * them. The path variation for those two cases occur inside the ParserUtil, and
+ * therefore should be covered by the ParserUtilTest.
+ */
+public class CheckinCommandParserTest {
+
+    private CheckInCommandParser parser = new CheckInCommandParser();
+
+    @Test
+    public void parse_validArgs_returnsCheckInCommand() {
+        RoomNumber expectedRoomNumber = new RoomNumber(VALID_ROOM_NUMBER_001);
+        assertParseSuccess(parser, ROOM_DESC_001, new CheckInCommand(expectedRoomNumber));
+    }
+
+    @Test
+    public void parse_invalidArgs_throwsParseException() {
```

Comment:
Might want to add comments on what the test cases mean

Reply:
noted. will do so in the next PR